src/main/java/org/opensearch/geospatial/ip2geo/dao/Ip2GeoCachedDao.java

```diff
@@ -112,9 +118,19 @@ private void remove(final String datasourceName) {
         getMetadata().remove(datasourceName);
     }
 
+    @Override
+    public void postIndex(ShardId shardId, Engine.Index index, Exception ex) {
+        log.warn("Skipped updating datasource metadata for datasource {} due to an indexing exception.", index.id(), ex);
```

Comment:
Good catch. If the index is datasource metadata, can we clear the cache to be safe?